From 5c4074afd5c18b96133872c09c81382dd46717ce Mon Sep 17 00:00:00 2001
From: Jason Perlow <jperlow@gmail.com>
Date: Mon, 12 Oct 2026 21:47:15 -0400
Subject: [PATCH] build: keep libcups out of the Vala target

dependency('cups') on the Vala library makes meson pass --pkg cups to valac,
which fails where valac ships no cups.vapi (Debian). Only the plain C
ipp_transport.c uses libcups, so build it as a static library with the cups
dependency and link it whole into libsingularity.

---
 meson.build | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index bd12f8b..500a58c 100644
--- a/meson.build
+++ b/meson.build
@@ -36,7 +36,10 @@ gst_app_dep    = dependency('gstreamer-app-1.0')
 libsingularity_deps = [gtk4_dep, wayland_dep, gee_dep, json_dep, peas_dep, gmodule_dep, gio_unix_dep, soup_dep, gtksource_dep, enchant_dep, m_dep]
 libsingularity_deps += [gst_dep, gst_video_dep, gst_app_dep]
 cups_dep       = dependency('cups')
-libsingularity_deps += [cups_dep]
+ipp_transport_lib = static_library('ipp-transport', 'src/print/ipp_transport.c',
+  dependencies: [dependency('glib-2.0'), cups_dep],
+  pic: true,
+)
 lcms2_dep      = dependency('lcms2')
 lcms2_c_args   = ['-I' + lcms2_dep.get_variable(pkgconfig: 'includedir')]
 lcms2_link_args = ['-L' + lcms2_dep.get_variable(pkgconfig: 'libdir'), '-llcms2']
@@ -382,7 +385,6 @@ libsingularity = shared_library('singularity',
   libsingularity_sources + libsingularity_resources + [
     'src/wayland/surface_effect.c',
     'src/wayland/snap_button.c',
-    'src/print/ipp_transport.c',
     'src/share/live_tls.c',
     'src/util/qr_decode.c',
     effect_protocol_c,
@@ -394,6 +396,7 @@ libsingularity = shared_library('singularity',
   dependencies: libsingularity_deps + [dependency('gnutls')],
   c_args: parental_c_args + lcms2_c_args,
   link_args: lcms2_link_args,
+  link_whole: ipp_transport_lib,
   vala_args: libsingularity_vala_args,
   include_directories: include_directories('src'),
   vala_header: 'singularity.h',
-- 
2.53.0

